src/frontend/src/App.js

```diff
@@ -33,42 +42,58 @@ function App() {
   // Fetch state of externalUserGames for multiplayer when the current user submits a guess
   // (the game state in the CurrentUserGame component changes)
   useEffect(() => {
-    if (gameState != null && gameState.metadata.gameID != null && isMultiplayerEnabled) {
+    if (
+      gameState != null &&
+      gameState.metadata.gameID != null &&
+      isMultiplayerEnabled
+    ) {
       const fetchNewExternalUserGames = async () => {
         try {
           const res = await getExternalUserGamesApi(gameState.metadata.gameID);
           setExternalGamesState(res);
+        } catch (e) {
+          setError("Failed to retrieve external user games" + e);
         }
-        catch (e) {
-          setError("Failed to retrieve external user games" + e)
-        }
-      }
+      };
 
       fetchNewExternalUserGames();
     }
-  }, [gameState, isMultiplayerEnabled])
+  }, [gameState, isMultiplayerEnabled]);
 
-  // TODO make this component the landing page for two separate pages
-  // 1) A single player page where CurrentUserGame is the only component (with Vassle header still)
-  // 2) A multiplayer page where the CurrentUserGame and x amount of ExternalUserGame components
-  // are on the same page
   return (
-    <>
-      <h1>Vaasle</h1>
-      <div className="App">
-        {error != null && <ErrorBadge text={error} />}
-        <div className="CurrentUserGame">
-          <CurrentUserGame gameState={gameState} setGameState={setGameState} />
-        </div>
-        {isMultiplayerEnabled && isAuthenticated && externalGamesState != null &&
-          Array.from(externalGamesState.externalUserGamesMap.values()).map((game, index) =>
-            <div className={`ExternalUserGame ${index % 2 === 0 ? 'odd' : 'even'}`}>
-              <ExternalUserGame externalUserGameGuesses={game} />
-            </div>
-          )
-        }
-      </div>
-    </>
+    <Router>
+      <Routes>
+        <Route path="/" element={<GameMode />} />
+        <Route
+          path="/singleplayer"
+          element={
+            <Singleplayer gameState={gameState} setGameState={setGameState} />
```

Comment:
We need to call the` initalGameState()` call back function here so that we're not just creating a new game each time the user logs in or out